parsec/backend/postgresql/user.py

```diff
@@ -46,6 +48,9 @@ async def create_user(
         self, organization_id: OrganizationID, user: User, first_device: Device
     ) -> None:
         async with self.dbh.pool.acquire() as conn:
```

Comment:
A transaction block is missing here

In fact the transaction block is already surrounding the `query_create_user` (that's what does the `@query(in_transaction=True)` decorator)

So I think the user limit check should be pushed into `query_create_user` 

The rational here is:
- `PGUserComponent` only implements the expected interface and calling `query_xxx`. The connection object is also created here given we want the `query_xxx` functions to be stateless.
- `query_xxx` is the actual implementation of the feature. It sets up the transaction context (given a feature operation should atomic)
- `query_xxx` makes use of multiple query functions (such as `send_signal` and all the `_q_yyy` instances of `Q` class), those query function should be combined together so they require a connection object and never deal with the transaction state